### PR TITLE
Don't expose Viewbox container as logical child.

### PR DESCRIPTION
**Pull request by grokys**
## What does the pull request do?

#7735 introduced an internal container control which hosts the child of `Viewbox`, but it exposed this child in the logical tree, breaking any styles which relied on the `Viewbox.Child` being the logical child of the `Viewbox`.

Fix this by introducing a simple internal `ViewboxContainer` control as the container.

cc: @amwx 

## Fixed issues

Fixes second part of #8092

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0020361-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Controls/Viewbox.cs [hunk 1]
```diff
@@ -88,7 +87,22 @@ protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs chang
...
             {
+                var (oldChild, newChild) = change.GetOldAndNewValue<IControl>();
+
+                if (oldChild is not null)
+                {
+                    ((ISetLogicalParent)oldChild).SetParent(null);
+                    LogicalChildren.Remove(oldChild);
+                }
+
                 _containerVisual.Child = change.GetNewValue<IControl>();
```
Comment (maxkatz6) [inline src/Avalonia.Controls/Viewbox.cs:98]:
```suggestion
                _containerVisual.Child = newChild;
```

File: src/Avalonia.Controls/Viewbox.cs [hunk 2]
```diff
@@ -129,5 +143,28 @@ protected override Size ArrangeOverride(Size finalSize)
 
             return finalSize;
         }
+
+        private class ViewboxContainer : Control
```
Comment (robloo) [inline src/Avalonia.Controls/Viewbox.cs:147]: Would be good IMO to document this control and why it's needed directly in code.

File: src/Avalonia.Controls/Viewbox.cs (hunk 1 above)
Comment (grokys) [reply src/Avalonia.Controls/Viewbox.cs]: Good catch @maxkatz6 !

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0020374-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by MarchingCube — APPROVED** [review]
LGTM. I guess adding a comment explaining why container is not a logical child could be useful.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0020422-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Controls/Viewbox.cs (hunk 2 above)
Comment (robloo) [reply src/Avalonia.Controls/Viewbox.cs:152]: Thanks!

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0020437-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by danielmayost** [comment]
@grokys This PR broke the Inheritance hierarchy from ViewBox to his kids, causing serious problems with FlowDirection behavior and I guess in other things.
Can you fix that?

Example:
image

**Pull request merged**